src/audio/base_fw.c

```diff
@@ -222,11 +222,13 @@ static int basefw_mem_state_info(uint32_t *data_offset, char *data)
 	index = 0;
 	tuple_data[index++] = info.free_phys_mem_pages;
 	tuple_data[index++] = info.ebb_state_dword_count;
+	for (i = 0; i < info.ebb_state_dword_count; i++) {
 #ifdef INTEL_ADSP
-	tuple_data[index++] = io_reg_read(LSPGCTL);
+		tuple_data[index + i] = io_reg_read(SHIM_LSPGCTL(i));
 #else
-	tuple_data[index++] = 0;
+		tuple_data[index + i] = 0;
 #endif
+	}
 	tuple_data[index++] = info.page_alloc_struct.page_alloc_count;
 	ptr = (uint16_t *)(tuple_data + index);
```

Comment:
this of course changes the behaviour significantly. Before this commit `index` was incremented twice in lines 226 and 228, now it isn't any more. Is this intended? Was that a bug?

Reply:
The tuple_data should contain the correct number of LPSRAM PGCTL register values (in the same way as it is done for HPSRAM). In the previous solution, if the number of ebb_state_dword_count equaled 0, a dword was added in tuple_data with the PGCTL status value for a non-existing LPSRAM bank. 

Therefore, the parser that is using data from tuple_data should not expect an "extra" LSPGCTL status value in the tuple_data, and then it could misinterpret it as some other type of data or status.

Reply:
@dnikodem right, but if `ebb_state_dword_count != 0`? Then you'd add several register values into `tuple_data` without incrementing `index`, so after the loop you'd overwrite the first of those values... I think you actually wanted to do
```
	for (i = 0; i < info.ebb_state_dword_count; i++) {
#ifdef INTEL_ADSP
		tuple_data[index++] = LPSRAM_REGS(i)->USxPGCTL;
#else
		tuple_data[index++] = 0;
#endif
	}
	if (info.ebb_state_dword_count)
		tuple_data[index++] = info.page_alloc_struct.page_alloc_count;
```

Reply:
@lyakh Fixed, thanks